File: Algorithms/Binary Search/binary_search.py
def binary_search(numbers=[], target=0):

    low = 0
    high = len(numbers)

    numbers.sort()

    while high >= low:

        mid = (low + high) // 2

        if numbers[mid] == target:
            return mid

        elif numbers[mid] > target:
            high = mid - 1

        else:
            low = mid + 1

    return -1

File: Algorithms/Binary Search/binarySearch.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int binarySearch(int *numbers, int sz, int target)
{

    int low = 0;
    int high = sz - 1;

    sort(numbers, numbers + sz);

    while (high >= low)
    {

        int mid = (high + low) / 2;

        if (numbers[mid] == target)
            return mid;

        else if (numbers[mid] > target)
            high = mid - 1;

        else
            low = mid + 1;
    }

    return -1;
}